File: src/C#/BehavioralPatterns/Mediator/CommunicationHub/Participants/Participant.cs
using CommunicationHub.Mediators;
using System;
using System.Collections.Generic;
using System.Text;

namespace CommunicationHub.Participants
{
    public abstract class Participant
    {
        public IMediator Mediator { get; set; }

        public abstract void Receive(Participant sender, object args);
    }
}

File: src/JavaScript/BehavioralPatterns/Mediator/CommunicationHub/participants/participant.js

class Participant {
  constructor() {
    this.mediator = null;
  }

  get mediator() {
    return this._mediator;
  }

  set mediator(value) {
    this._mediator = value;
  }

  receive(sender, args) {
    throw new Error("Method 'receive' must be implemented.");
  }
}

module.exports = Participant;
